### ISSUE TITLE
vulkaninfo cannot get info on GPU 1 only

### ISSUE THREAD

**Issue by alexpanter** [KhronosGroup/Vulkan-Tools#840]
Hi,

Unless I completely misunderstand how to use the `vulkaninfo` tool on the command line, there's something strange afoot here..
If I run the tool without any arguments, I get a list like this:
```
Layers: count = 4
=================
VK_LAYER_INTEL_nullhw (INTEL NULL HW) Vulkan version 1.1.73, layer version 1:
	Layer Extensions: count = 0
	Devices: count = 3
		GPU id = 0 (AMD Radeon Graphics (RADV REMBRANDT))
		Layer-Device Extensions: count = 0

		GPU id = 1 (NVIDIA GeForce RTX 3070 Ti Laptop GPU)
		Layer-Device Extensions: count = 0

		GPU id = 2 (llvmpipe (LLVM 15.0.7, 256 bits))
		Layer-Device Extensions: count = 0

[...]
```

According to the help message (`vulkaninfo --help`) I should be able to get info on a select GPU:
```
[-j=<gpu-number>, --json=<gpu-number>]
                    For a multi-gpu system, a single gpu can be targetted by
                    specifying the gpu-number associated with the gpu of 
                    interest. This number can be determined by running
                    vulkaninfo without any options specified.
```

And then I try to run it like this:
```
vulkaninfo -j=1
```

-- but I only get the help message, alluring the fact that I'm possibly using the tool in the wrong way. And I do not want a JSON output, but rather text, which is more easily readable.

Platform is Ubuntu 23.04.

**Comment by charles-lunarg** [comment]
Just tested this locally, this does look like a regression.

I can say that the -j output only works for the json output, but in retrospect that is a silly limitation that should be fixed (so there should be a way to choose the gpu for text/html).

**Comment by alexpanter** [comment]
Absolutely, if that could work for text format as well it would be great!

And a small sidenote: If this could also be implemented with `--show-formats` it would become even more helpful. :)

**Comment by charles-lunarg** [comment]
I have fixed the bug, but have not added support for 'select a gpu in text/html output'. I'll remove the bug tag to reflect this change.
